src/cli/restore/onedrive_test.go

```diff
@@ -50,3 +51,36 @@ func (suite *OneDriveSuite) TestAddOneDriveCommands() {
 		})
 	}
 }
+
+func (suite *OneDriveUnitSuite) TestAddOneDriveCommands_hasRestoreFlag() {
+	expectUse := oneDriveServiceCommand + " " + oneDriveServiceCommandUseSuffix
+
+	table := []struct {
+		name        string
+		use         string
+		expectUse   string
+		expectShort string
+		expectRunE  func(*cobra.Command, []string) error
+	}{
+		{"restore onedrive", restoreCommand, expectUse, oneDriveRestoreCmd().Short, restoreOneDriveCmd},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			cmd := &cobra.Command{Use: test.use}
+
+			c := addOneDriveCommands(cmd)
+			require.NotNil(t, c)
+
+			cmd.SetArgs([]string{"onedrive", "--restore-permissions"})
+
+			actual := new(bytes.Buffer)
+			cmd.SetOut(actual)
+			cmd.SetErr(actual)
+			cmd.Execute()
+
+			assert.NotContains(t, actual.String(), "Error: unknown flag: --restore-permissions")
```

Comment:
Not sure if this test is too hacky. I don't know if there is a simpler way to test this with our current setup.